### DISCUSSION TITLE
Can't use filament pkg in react

### DISCUSSION THREAD

**Discussion by vinkovsky** [google/filament#5106] [Q&A]
Hey! I'm using nextjs framework but react is under the hood. When i try to import filament pkg in my react component i get this error

`Module not found: cannot resolve 'fs'`

Nextjs provides a webpack config but I have no idea how to set it up to solve this problem.

**Comment by lttkachenko** [comment]
Yes they don't support any bundler for `Filament` so it won't work inside `React` or any other frameworks that use bundler like webpack to assemble application.
Moreover this, `NextJS` is `SSR` (server-side rendering) framework and this makes any component that uses window object (and Filament does, at least as said in their single tutorial) useless, cause you will have another compile time error: `Window object is not defined`.
Also you cannot use `Filament` out of `CDNs` cause minified version does not contain the necessary static types and properties like `Filament.MinFilter.LINEAR_MIPMAP_LINEAR`.
Even if you'll try using `next-dynamic` to use it with `{ ssr: false }` setting and put `'use client';` to your component.

Google team provides another ready to use component - Google Model Viewer which works with React and possibly NextJS.
My sorrow is that component does not support moving model across the scene and rendering several models inside single scene, just rendering single model in 0-0-0 point and trackball rotation/zoom.
